src/Veldrid.Utilities/MtlParser.cs

```diff
@@ -91,78 +91,99 @@ public void Process(string line)
                 switch (pieces[0].ToLowerInvariant().Trim())
                 {
                     case "newmtl":
-                        ExpectExactly(pieces, 1, "newmtl");
+                        ExpectAtLeast(pieces, 1, "newmtl");
                         FinalizeCurrentMaterial();
-                        _currentDefinition = new MaterialDefinition(pieces[1]);
+                        if (!DictMatDef.ContainsKey(pieces[1]))
```

Comment:
There's a lot of diff here just because `_currentDefinition` might be null. I think it would be a lot simpler if you just did the duplication check over in `FinalizeCurrentMaterial`. IMO, you can also just get rid of the Dictionary altogether, and loop through the other materials and check their names in that method. There are going to be such a small number of materials in the majority of cases that it probably won't be any faster to use a Dictionary.